src/orquestra/sdk/_ray/_dag.py

```diff
@@ -862,7 +842,12 @@ def get_workflow_run_outputs_non_blocking(
                 f"{workflow_run_id} has not succeeded. {workflow_state}",
                 workflow_state,
             )
-        return self.get_workflow_run_outputs(workflow_run_id)
+        try:
+            return self._client.get_workflow_output(workflow_run_id)
+        except ValueError as e:
+            raise exceptions.NotFoundError(
+                f"Workflow run {workflow_run_id} wasn't found"
+            ) from e
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏These are uncovered, can you add a test to make sure this happens?

Reply:
It was a case that's unlikely to happen in normal circumstances. I simplified it in bad2622